### ISSUE TITLE
MDX Stories with Knobs

### ISSUE THREAD

**Issue by benwolfram** [storybookjs/storybook#10066]
**Describe the bug**
While knobs are enabled to supply prop values, having multiple stories within an MDX stories file using the same component more than once causes the components to share identical values. Working correctly in Canvas tab when stories are displayed individually. Also is corrected when removing knobs and hard-coding values.

**To Reproduce**
I have an example repo set up here [https://github.com/benwolfram/mdx-knobs-multi-stories](https://github.com/benwolfram/mdx-knobs-multi-stories)

**Expected behavior**
For the stories to be displayed given their default knob props values given when in Docs view.

**Screenshots**
![Displayed in docs view](https://github.com/benwolfram/mdx-knobs-multi-stories/raw/master/src/img/docs.png)

![Displayed in canvas view](https://github.com/benwolfram/mdx-knobs-multi-stories/raw/master/src/img/canvas.png)

**Code snippets**
```
import { Meta, Preview, Story, Props, Source, SourceState } from '@storybook/addon-docs/blocks';
import { number, select } from "@storybook/addon-knobs";
import { CircularProgress } from '@material-ui/core';

<Meta title="CircularProgress" component={CircularProgress} />

# Circular Progress

<Preview>
  <Story name="Standard">
    <CircularProgress />
  </Story>
</Preview>

<Props of={CircularProgress} />

### Thickness

<Preview withSource={SourceState.NONE}>
  <Story name="Thick">
    <CircularProgress thickness={number("Thickness", 5)} size={number("Size", 100)} />
  </Story>
  <Story name="Thicker">
    <CircularProgress thickness={number("Thickness", 10)} size={number("Size", 100)} />
  </Story>
  <Story name="Thickest">
    <CircularProgress thickness={number("Thickness", 15)} size={number("Size", 100)} />
  </Story>
</Preview>
```

**System:**
Environment Info:
  System:
    OS: macOS 10.15.2
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 13.8.0 - /usr/local/bin/node
    Yarn: 1.22.0 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.122
    Firefox: 73.0.1
    Safari: 13.0.4
  npmPackages:
    @storybook/addon-docs: ^5.3.14 => 5.3.14 
    @storybook/addon-knobs: ^5.3.14 => 5.3.14 
    @storybook/addons: ^5.3.14 => 5.3.14 
    @storybook/react: ^5.3.14 => 5.3.14 
    @storybook/source-loader: ^5.3.14 => 5.3.14

**Comment by shilman** [comment]
Hey, we're working on getting this fixed as part of 6.0! Hang tight!

**Comment by danawoodman** [comment]
@shilman is this on the roadmap? Would be super cool to have knobs in the MDX stories!

**Comment by shilman** [comment]
Coming in 6.0 https://github.com/storybookjs/storybook/pull/10834

**Comment by shilman** [comment]
Hi gang, We’ve just released addon-controls in 6.0-beta!

Controls are portable, auto-generated knobs that are intended to replace addon-knobs long term.

- [Controls README](https://github.com/storybookjs/storybook/tree/next/addons/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try them out today. Thanks for your help and support getting this stable for release!

**Comment by benwolfram** [comment]
Woot woot 🎉 Jus seeing this now, I will try things out on Monday!

**Comment by shilman** [comment]
For anybody who is interested in Controls but don't know where to start, I've created a quick & dirty step-by-step walkthrough to go from a fresh CRA project to a working demo. Check it out:

=> [Storybook Controls w/ CRA & TypeScript](https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487)

There are also some "knobs to controls" migration docs in the Controls README:

=> [How do I migrate from addon-knobs?](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

**Comment by danawoodman** [comment]
@shilman just tried it out and it's great! Very excited to see controls supported in MDX stories! 💯 👏

**Comment by shilman** [comment]
Thanks @danawoodman ❤️

**Comment by JGJP** [comment]
This issue has been closed, but as detailed in #11984 the dynamic part of knobs (important functionality for some people) hasn't been implemented in controls, so while the specific problem of the author has been solved, it's my understanding that we still can't be said to have knobs in mdx. Is this correct?
